allennlp/data/iterators/data_iterator.py

```diff
@@ -225,19 +225,35 @@ def _memory_sized_lists(self,
             yield list(iterator)
 
 
-    def _ensure_batch_is_sufficiently_small(self, batch_instances: Iterable[Instance]) -> List[List[Instance]]:
+    def _ensure_batch_is_sufficiently_small(
+            self,
+            batch_instances: Iterable[Instance],
+            excess: deque) -> List[List[Instance]]:
```

Comment:
For everything but core types, it's better to use the collection's corresponding entry in the `typing` module, because it allows you to specify what the collection is of, e.g:

```
from typing import Deque

...
excess: Deque[Instance]) -> ....
```
It's also less confusing, because they are strictly abstract, so it's obvious they're being used for type hints or inheritance.

Reply:
Good call, thanks! I didn't know that all collection classes had corresponding types.